src/steamship/agents/service/agent_service.py

```diff
@@ -29,15 +29,26 @@ def run_action(self, action: Action, context: AgentContext):
                 "Please use synchronous Tasks (Tools that return List[Block] for now."
             )
         else:
+            outputs = ",".join([f"{b.as_llm_input()}" for b in blocks_or_task])
+            logging.info(
+                f"Tool {action.tool.name}: ({outputs})",
+                extra={
+                    AgentLogging.TOOL_NAME: action.tool.name,
+                    AgentLogging.IS_MESSAGE: True,
+                    AgentLogging.MESSAGE_TYPE: AgentLogging.OBSERVATION,
+                    AgentLogging.MESSAGE_AUTHOR: AgentLogging.AGENT,
+                },
+            )
             action.output = blocks_or_task
             context.completed_steps.append(action)
 
     def run_agent(self, agent: Agent, context: AgentContext):
         action = agent.next_action(context=context)
         while not isinstance(action, FinishAction):
             # TODO: Arrive at a solid design for the details of this structured log object
+            inputs = ",".join([f"{b.as_llm_input()}" for b in action.input])
             logging.info(
-                f"Running Tool {action.tool.name}",
+                f"Running Tool {action.tool.name} ({inputs})",
```

Comment:
Does this push user data to our log system?

Reply:
potentially. the text of whatever the agent looks at, at least. so, when a tool is called with user input directly, then yes?